apps/chat/src/store/share/share.epics.ts

```diff
@@ -476,49 +473,242 @@ const getSharedListingSuccessEpic: AppEpic = (action$, state$) =>
-            }),
-          );
-          actions.push(
-            PromptsActions.addFolders({
-              folders: payload.resources.folders.map((res) => ({
-                ...res,
-                sharedWithMe: true,
-              })) as FolderInterface[],
-            }),
-          );
+          payload.resources.entities.length &&
+            actions.push(
+              PromptsActions.addPrompts({
+                prompts: payload.resources.entities.map((res) => ({
+                  ...res,
+                  sharedWithMe: true,
+                })) as Prompt[],
+              }),
+            );
+          payload.resources.folders.length &&
+            actions.push(
+              PromptsActions.addFolders({
+                folders: payload.resources.folders.map((res) => ({
+                  ...res,
+                  sharedWithMe: true,
+                })) as FolderInterface[],
+              }),
+            );
         }
       }
 
       return concat(actions);
     }),
   );
 
+const revokeAccessEpic: AppEpic = (action$) =>
+  action$.pipe(
+    filter(ShareActions.revokeAccess.match),
+    switchMap(({ payload }) => {
+      const resourceUrl =
+        payload.nodeType === BackendDataNodeType.FOLDER
+          ? encodeApiUrl(payload.resourceId) + '/'
+          : encodeApiUrl(payload.resourceId);
+
+      return ShareService.shareRevoke([resourceUrl]).pipe(
+        switchMap(() => {
+          return of(ShareActions.revokeAccessSuccess(payload));
+        }),
+        catchError(() => {
+          return of(
+            UIActions.showErrorToast(
+              translate(errorsMessages.revokeAccessFailed),
+            ),
+          );
+        }),
+      );
+    }),
+  );
+
+const revokeAccessSuccessEpic: AppEpic = (action$) =>
+  action$.pipe(
+    filter(ShareActions.revokeAccessSuccess.match),
+    switchMap(({ payload }) => {
+      if (
+        payload.nodeType === 'ITEM' &&
+        payload.resourceType === BackendResourceType.CONVERSATION
+      ) {
+        return of(
+          ConversationsActions.updateConversationSuccess({
+            id: payload.resourceId,
+            conversation: {
+              isShared: false,
+            },
+          }),
+        );
+      }
+      if (
+        payload.nodeType === 'FOLDER' &&
+        payload.resourceType === BackendResourceType.CONVERSATION
+      ) {
+        return of(
+          ConversationsActions.updateFolder({
+            folderId: payload.resourceId,
+            values: {
+              isShared: false,
+            },
+          }),
+        );
+      }
+      if (
+        payload.nodeType === 'ITEM' &&
+        payload.resourceType === BackendResourceType.PROMPT
+      ) {
+        return of(
+          PromptsActions.updatePromptSuccess({
+            id: payload.resourceId,
+            prompt: {
+              isShared: false,
+            },
+          }),
+        );
+      }
+      if (
+        payload.nodeType === 'FOLDER' &&
```

Comment:
```suggestion
        payload.nodeType === BackendDataNodeType.FOLDER &&
```
Please, also check all the places below